src/taker/api.rs

```diff
@@ -472,6 +497,12 @@ impl Taker {
             }
         }
 
+        // Sweep the incoming swapcoins to an internal wallet address.
+        if !self.ongoing_swap_state.incoming_swapcoins.is_empty() {
+            let feerate = 2f64; // Hardcoded feerate value
+            self.sweep_incoming_swapcoins(feerate)?;
+        }
```

Comment:
This should be an `assert!()` instead of conditional. Ensure that incoming_swapcoin at this stage is never empty. Then sweep it. 

Its logically equivalent, but thats a more clear code with clear intention.